Scripts/zombie_head.py

```diff
@@ -0,0 +1,34 @@
+import pygame
+import time
+
+class ZombieHead(pygame.sprite.Sprite):
+    """
+    The ZombieHead class represents a zombie object in the game.
+    It can be spawned, smashed by the player, and automatically despawned after a set amount of time.
+    """
+
+    def __init__(self, spawn_data):
+        """
+        Initialize the ZombieHead with its image, sound, and spawn location.
+        """
+        super().__init__()
+        self.image = pygame.image.load('img/zombie_head.png')  # Optimized for performance with convert()
+        self.rect = self.image.get_rect()
+        self.kill_audio = pygame.mixer.Sound('audio/punch.wav')
+        self.rect.topleft = spawn_data.spawn_point()  # Set spawn location
```

Comment:
ông có thể chỉnh nó spawn ở giữa tâm thay vì ngay góc để hợp với mấy tọa độ spawn hơn:
        # set spawn zombie in center
        x,y = spawn_data.spawn_point()
        x = x - self.rect.width/2
        y = y - self.rect.height/2
        self.rect.topleft = (x,y) # Set spawn location

Reply:
hoặc là self.rect.center là gọn nhất

Reply:
self.image = pygame.image.load('img/zombie_head.png').convert_alpha()  # Load image with transparency
        self.rect = self.image.get_rect()
        self.kill_audio = pygame.mixer.Sound('audio/punch.wav')
        self.rect.center = spawn_data.spawn_point()